extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTest.java

```diff
@@ -2461,6 +2461,20 @@ private boolean isTaskReading(KinesisIndexTask task)
     return task.getRunner().getStatus() == SeekableStreamIndexTaskRunner.Status.READING;
   }
 
+  /**
+   * Return true if specified task is in READING OR PUBLISHING state
+   * This helper method allows UTs to wait until the task associated with the test has started before continuing on with
+   * the test. Checking for PUBLISHING state in addition to READING prevents a race condition where a task could go
+   * publishing before the wait for READING state had started.
+   * @param task {@link KinesisIndexTask} having its state checked
+   * @return true if task is in READING or PUBLISHING state, otherwise false
+   */
+  private boolean isTaskReadingOrPublishing(KinesisIndexTask task)
```

Comment:
If there are race conditions, do we really need to wait for READING state at all?
I wonder if simply waiting on PUBLISHING state would make the tests easier to understand.

Reply:
Fair point, I'll investigate

Reply:
The three instances where I moved the test to `isTaskReadingOrPublishing` seem to me to not care at all about the details of if it is reading, publishing, or both. so I believe it is safe to change this new method simply to `isTaskPublishing` and wait on that if it makes things less confusing. Will put it through the paces locally and then push a change if I confirm